web/webkit/src/main/resources/toserve/lift.js

```diff
@@ -76,8 +76,12 @@
       },
       cometGetTimeout: 140000,
       cometFailureRetryTimeout: 10000,
-      cometOnSessionLost: function() {
-        window.location.href = "/";
+      cometOnSessionLost: function(contextPath) {
+        if (contextPath === null || contextPath === undefined) {
+          window.location.href = "/";
+        } else {
+          window.location.href = contextPath;
```

Comment:
This could really just be simplified to `window.location.href = contextPath || "/"`.